data-model/src/main/java/io/micronaut/data/model/query/builder/sql/Dialect.java

```diff
@@ -143,4 +159,13 @@ public final boolean supportsJoinType(@NonNull Join.Type joinType) {
         return this.joinTypesSupported.contains(joinType);
     }
 
+    /**
+     * Gets an indicator whether JSON entity is supported by the database.
```

Comment:
What exactly is JSON entity? I think we want to know if DB supports JSON operations on a JSON column